library/Vanilla/EmbeddedContent/Factories/AbstractOwnSiteEmbedFactory.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * @author Adam Charron <adam.c@vanillaforums.com>
+ * @copyright 2009-2019 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+namespace Vanilla\EmbeddedContent\Factories;
+
+use Garden\Web\Exception\NotFoundException;
+use Garden\Web\RequestInterface;
+use Vanilla\Contracts\Site\SiteSectionProviderInterface;
+use Vanilla\EmbeddedContent\AbstractEmbed;
+use Vanilla\EmbeddedContent\AbstractEmbedFactory;
+use Vanilla\EmbeddedContent\Embeds\QuoteEmbed;
+use Vanilla\Site\SiteSectionModel;
+use Vanilla\Web\Asset\SiteAsset;
+
+/**
+ * Embed factory with utilities for matching our own site.
+ */
+abstract class AbstractOwnSiteEmbedFactory extends AbstractEmbedFactory {
```

Comment:
Why do we need this class to  be `abstract`?

Reply:
Because it doesn't implement all methods from `AbstractEmbedFactory`. Notably `getSupportedPathRegex()`